frontend/src/containers/EngagementView/EngagementViewContainer.js

```diff
@@ -24,27 +25,42 @@ class EngagementViewContainer extends Component<Props, State> {
     isError: false,
     service: 0,
     civilMil: 0,
+    noShows: 0,
     requirements: []
   }
 
+  setErrorState = () => {
+      this.setState({
+          service: 0,
+          civilMil: 0,
+          noShows: 0,
+          requirements: [],
+          isLoading: false,
+          isError: true,
+      })
+  }
+
   componentDidMount () {
     getRequirements(this.props.student.id)
       .then((data) => {
         this.setState({
           service: data.acceptedService,
           civilMil: data.acceptedCivilMil,
           requirements: data.requirements,
-          isLoading: false,
         })
+        getAttendance(this.props.student.id)
```

Comment:
Is it possible to un-nest these? They don't depend on each other, so may be slowing down the app to only start the second request upon completion of the first.